src/sql/src/plan/statement/ddl.rs

```diff
@@ -2992,27 +3016,98 @@ pub fn plan_create_connector(
+                        let username = opts_map.remove("USERNAME").expect("USERNAME is expected if SECURITY = SASL-SSL").expect("USERNAME must have a value if SECURITY is SASL-SSL").to_string();
+                        let password = resolve_and_track_option( &mut opts_map, "PASSWORD")?.expect("PASSWORD is required if SECURITY = SASL-SSL");
+                        KafkaSecurityOptions::SASL { mechanism, ssl: true, username, password }
+                    }
+                    "SASL-PLAIN" => {
+                        let mechanism = opts_map.get("MECHANISM").expect("MECHANISM is expected if SECURITY = SASL-PLAIN").expect("MECHANISM is required if SECURITY = SASL-PLAIN").to_string();
+                        let username = opts_map.get("USERNAME").expect("USERNAME is expected if SECURITY = SASL-PLAIN").expect("USERNAME is required if SECURITY = SASL-PLAIN").to_string();
+                        let password = resolve_and_track_option(&mut opts_map, "PASSWORD")?.expect("PASSWORD is required if SECURITY = SASL-PLAIN");
+                        KafkaSecurityOptions::SASL { mechanism, ssl: false, username, password }
+                    }
+                    "PLAINTEXT" => {
+                        KafkaSecurityOptions::PLAINTEXT
+                    }
+                    _ => bail!("Unsupported value for SECURITY, must be one of: SSL, SASL-SSL, SASL-PLAIN, or PLAINTEXT")
+                }
+                None => KafkaSecurityOptions::PLAINTEXT,
+                _ => bail!("Invalid SECURITY parameters"),
+            };
+            if !opts_map.is_empty() {
+                bail!(
+                    "invalid config: unused options: {}",
+                    opts_map.keys().join(", ")
+                );
+            }
             ConnectorInner::Kafka {
                 broker: broker.parse()?,
-                config_options: kafka_util::extract_config(&mut with_options)?,
+                security,
             }
         }
-        CreateConnector::CSR {
-            registry,
-            with_options,
-        } => {
-            let with_options = normalize::options(&with_options);
+        CreateConnector::CSR { registry, security } => {
+            let mut opts_map = BTreeMap::from_iter(
+                security
+                    .iter()
+                    .map(|o| (o.key.to_string().to_uppercase(), o.value.as_ref())),
+            );
+            let password = resolve_and_track_option(&mut opts_map, "PASSWORD")?;
+            let authority = resolve_and_track_option(&mut opts_map, "AUTHORITY")?;
+            let certificate = resolve_and_track_option(&mut opts_map, "CERTIFICATE")?;
+            let key = resolve_and_track_option(&mut opts_map, "KEY")?;
+            let username =
+                if let Some(Some(WithOptionValue::Value(val))) = opts_map.remove("USERNAME") {
+                    // For some reason this sometimes comes in single quoted, if so then strip both ends
```

Comment:
I don't think "For some reason" is quite sufficient here--this sounds like an escape's occurring, and we need to know what's causing it rather than just trying to "cure the symptom."

Reply:
Agreed, my thought had been that if the parsing was going to be changing anyway that would be an opportune time to resolve this but I can also dive deeper into this since I couldn't immediately determine why `parse_literal_string` as the previous iteration of the parsing code used would behave differently than the `parse_value` that `parse_with_option_value` uses.